source/agora/test/SCPState.d

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+
+    SCPState Related Test
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.SCPState;
+
+version (unittest):
+
+import agora.common.Config;
+import agora.common.Serializer;
+import agora.common.Task;
+import agora.common.Hash;
+import agora.common.crypto.Key;
+import agora.consensus.protocol.Nominator;
+import agora.consensus.data.Block;
+import agora.consensus.data.ConsensusParams;
+import agora.consensus.data.Transaction;
+import agora.consensus.LatestEnvelopes;
+import agora.node.Ledger;
+import agora.node.Validator;
+import agora.network.Clock;
+import agora.network.NetworkManager;
+import agora.test.Base;
+
+import geod24.Registry;
+
+import scpd.Cpp;
+import scpd.types.Stellar_SCP;
+
+import core.stdc.time;
+import core.thread;
+
+/// A test to store SCP state and recover SCP state
+unittest
+{
+    // static variable for comparing values
+    static SCPEnvelope[] envelopes;
+
+    static class ReNominator : TestNominator
+    {
+        /// Ctor
+        public this (immutable(ConsensusParams) params, Clock clock,
+            NetworkManager network, KeyPair key_pair, Ledger ledger,
+            TaskManager taskman, string data_dir, ulong txs_to_nominate)
+        {
+            super(params, clock, network, key_pair, ledger, taskman, data_dir,
+                txs_to_nominate);
+        }
+
+        ///
+        override void restoreSCPState ()
+        {
+            uint i;
```

Comment:
You should add `assert(this.latest_envelopes.length > 0);` or `assert(i > 0)` after the foreach.

Reply:
Yes, there will be an error with the corresponding assert.
I'am looking for the cause.